quartermaster/src/main/java/com/github/reflectoring/infiniboard/quartermaster/dashboard/DashboardController.java

```diff
@@ -5,40 +5,54 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.github.reflectoring.haljson.HalJsonResource;
+import com.github.reflectoring.infiniboard.packrat.dashboard.DashboardConfig;
+import com.github.reflectoring.infiniboard.packrat.dashboard.DashboardConfigRepository;
+import com.github.reflectoring.infiniboard.packrat.widget.WidgetConfigRepository;
 
 import static org.springframework.http.HttpStatus.NOT_FOUND;
 import static org.springframework.http.HttpStatus.OK;
 import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
 
 @RestController
 @RequestMapping("/api/dashboards")
 public class DashboardController {
 
-    private DashboardRepository repository;
-    private DashboardMapper     dashboardMapper;
+    private DashboardConfigRepository dashboardConfigRepository;
+    private WidgetConfigRepository    widgetConfigRepository;
+    private DashboardMapper           dashboardMapper;
 
     @Autowired
-    public DashboardController(DashboardRepository repository, DashboardMapper dashboardMapper) {
-        this.repository = repository;
+    public DashboardController(DashboardConfigRepository dashboardConfigRepository,
+                               WidgetConfigRepository widgetConfigRepository,
+                               DashboardMapper dashboardMapper) {
+        this.dashboardConfigRepository = dashboardConfigRepository;
+        this.widgetConfigRepository = widgetConfigRepository;
         this.dashboardMapper = dashboardMapper;
     }
 
     @RequestMapping(method = GET)
-    public ResponseEntity<List<HalJsonResource>> getAllDashboardConfigurations() {
-        List<Dashboard>       dashboards = repository.findAll();
+    public ResponseEntity<List<HalJsonResource>> getAll() {
+        List<DashboardConfig> dashboards = dashboardConfigRepository.findAll();
+
+        dashboards.stream().map(this::addAllWidgetsToDashboard);
         List<HalJsonResource> resources  = dashboardMapper.toResources(dashboards);
 
         return new ResponseEntity<>(resources, OK);
     }
 
 
     @RequestMapping(value = "/{id}", method = GET)
-    public ResponseEntity<HalJsonResource> getDashboardConfiguration(@PathVariable int id) {
-        Dashboard dashboard = repository.find(id);
+    public ResponseEntity<HalJsonResource> getOne(@PathVariable String id) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUndocumentedApi)